src/bootstrap/src/core/config/config.rs

```diff
@@ -183,6 +184,62 @@ pub enum GccCiMode {
     DownloadFromCi,
 }
 
+/// This enum is used for deserializing change IDs from TOML, allowing both numeric values and the string `"ignore"`.
+#[derive(Clone, Debug, PartialEq)]
+pub enum ChangeId {
+    Ignore,
+    Id(usize),
+}
+
+impl<'de> Deserialize<'de> for ChangeId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct ChangeIdVisitor;
```

Comment:
This is a bit too complicated. You can do something like this instead (with some better error messages added):
```rust
let value = <&str>::deserialize(deserializer)?;
        if value == "ignore" {
            Ok(ChangeId::Ignore)
        } else {
            let id = value.parse()?;
            Ok(ChangeId::Id(id))
        }
```

Also, you can use `deserialize_with`, as we only deserialize the field on one place :)